library/std/src/sys_common/poison.rs

```diff
@@ -3,6 +3,9 @@ use crate::fmt;
 use crate::sync::atomic::{AtomicBool, Ordering};
 use crate::thread;
 
+#[cfg(doc)]
+use crate::sync::{Mutex, RwLock};
```

Comment:
This isn't a good idea in general because these links will not work after being inlined cross crate. Dependencies aren't built with `cfg(doc)` enabled.

Reply:
Ah hmm good point. Should it allow unused imports then?
```suggestion
#[allow(unused_imports)] // for intra-doc links
use crate::sync::{Mutex, RwLock};
```

Reply:
That means https://github.com/rust-lang/rust/pull/75705/files#diff-fb2d2f6cd15a292c22604bee27a7ab2fR61 should be changed too, I can make a PR for that.

Reply:
> Should it allow unused imports then?

That seems reasonable.